Document/0x05b-Basic-Security_Testing.md

```diff
@@ -74,10 +74,91 @@ As mentioned before, starting with Android 7, the Android OS will no longer trus
 
 #### Bypassing the Network Security Configuration
 
-From Android 7 onwards, the network security configuration allows apps to customize their network security settings, by defining which CA certificates the app will be trusting. The CA certificates trusted by the app can be a system trusted CA as well as a user CA.
-The network security configuration uses an XML file where the app specifies which CA certificates will be trusted.
+From Android 7 onwards, the network security configuration allows apps to customize their network security settings, by defining which CA certificates the app will be trusting. The apps must also include an entry in the manifest file to point to the network security configuration file.
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<manifest ... >
+    <application android:networkSecurityConfig="@xml/network_security_config"
+                    ... >
+        ...
+    </application>
+</manifest>
+```
+
+The network security configuration uses an XML file where the app specifies which CA certificates will be trusted. There are various ways to bypass the NSC, which will be described below.  
+
+##### Adding the User Certificates to the NSC Configuration
+
+The CA certificates trusted by the app can be a system trusted CA as well as a user CA. If you added the certificate of your interception proxy already as additional CA in Android, you can force the Android app to trust it with the following NSC configuration below:
+
+```xml
+<network-security-config>
+   <base-config>
+      <trust-anchors>
+          <certificates src="system" />
+          <certificates src="user" />
+      </trust-anchors>
+   </base-config>
+</network-security-config>
+```
+
+To implement this new setting you must follow the steps below:
+
+- Decompile the app using decompilation tools.[Manual static Analysis](https://github.com/OWASP/owasp-mstg/blob/master/Document/0x05b-Basic-Security_Testing.md#manual-static-analysis) provides details about decompiling
+- Make the application trust user's certificates by creating a network security configuration that includes `<certificates src="user" />` as explained above
+- Repackage the app. The [Android developer documentation](https://developer.android.com/studio/publish/app-signing#signing-manually) explains how it's done.
+
+Note that even if this method is quite simple its major drawback is that you have to apply this operation for each application you want to evaluate which is additional overhead for testing.
```

Comment:
Maybe add that applications can also check for resource integrity / signature and that this approach may fail for hardened apps?

Reply:
good point! Just added it